[DevOps] Fix Snyk project issue

ChathurindaRanasinghe opened the pull request.

pascal-pfeiffer commented on .github/workflows/snyk-scan.yml:
```diff
@@ -20,8 +24,16 @@ jobs:
         with:
           python-version: '3.10'
+      
+      - name: Check changed Deps files
+        uses: tj-actions/changed-files@v35
+        id: changed-files
+        with:
+          files: | # This will match all the files with below patterns
+            Pipfile
```
what is the intention here?

ChathurindaRanasinghe replied:
There might be scenario if some one make PR and although he does not add or remove any packages it still might fail due to previous issues. This is to avoid that.

pascal-pfeiffer commented on .github/workflows/snyk-scan.yml:
```diff
@@ -8,10 +8,14 @@ on:
 jobs:
   snyk_scan_test:
-    if: ${{ github.event_name == 'pull_request' }} || ${{ github.event_name == 'workflow_dispatch' }}
+    if: ${{ github.event_name == 'push' }} || ${{ github.event_name == 'workflow_dispatch' }}
```
only on push?
Why do we have "on" pull_request above then?

Could you please give a quick summary of what this workflow shall achieve?

ChathurindaRanasinghe replied:
That was only for testing purposes. It is fixed.

pascal-pfeiffer reviewed (commented):
Thank you!
Could you please give a quick summary of what this workflow shall achieve.

It seems to be still WIP (includes the devops branch and modified Pipfile). Is that true?

ChathurindaRanasinghe commented:
> Thank you! Could you please give a quick summary of what this workflow shall achieve.
> 
> It seems to be still WIP (includes the devops branch and modified Pipfile). Is that true?

Devops branch only used for testing purposes. It is fixed now.
I added checking for changed files and remove dev packages before scan.

pascal-pfeiffer reviewed (approved):
lgtm
